server/text_generation_server/models/custom_modeling/mpt_modeling.py

```diff
@@ -330,7 +348,15 @@ def __init__(
             config, prefix=f"{prefix}.Wqkv", weights=weights, bias=not config.no_bias
         )
         if self.qk_ln:
-            raise NotImplementedError("qk_ln is not supported")
+            if weights.process_group.size() > 1:
+                raise NotImplementedError("qk_ln is not supported for number of shards > 1")
+            bias = not config.no_bias
+            hidden_size = config.d_model
+            head_dim = hidden_size // self.n_heads
+
+            norm_class = LPLayerNorm
+            self.q_ln = norm_class(d_model, bias=bias, prefix=f"{prefix}.q_ln", weights=weights)
+            self.k_ln = norm_class(self.n_heads * head_dim, prefix=f"{prefix}.k_ln", weights=weights)
```

Comment:
```suggestion
            self.q_ln = LPLayerNorm(d_model, bias=bias, prefix=f"{prefix}.q_ln", weights=weights)
            self.k_ln = LPLayerNorm(self.n_heads * head_dim, prefix=f"{prefix}.k_ln", weights=weights)
```

Feels  a bit more direct.

Reply:
Yes, we can do away with the extra variable.